contracts/interchain-token-service/src/contract.rs

```diff
@@ -539,8 +523,59 @@ impl InterchainTokenService {
                     );
                 }
             }
-            Message::DeployInterchainToken(_) => {
-                todo!()
+            Message::DeployInterchainToken(DeployInterchainToken {
+                token_id,
+                name,
+                symbol,
+                decimals,
+                minter,
+            }) => {
+                ensure!(!name.is_empty(), ContractError::EmptyTokenName);
+
+                ensure!(!symbol.is_empty(), ContractError::EmptyTokenSymbol);
+
+                ensure!(
+                    !env.storage()
+                        .persistent()
+                        .has(&DataKey::TokenIdConfigKey(token_id.clone())),
+                    ContractError::TokenAlreadyDeployed
+                );
+
+                let token_metadata = TokenMetadata {
+                    name,
+                    symbol,
+                    decimal: decimals as u32,
+                };
+
+                let minter_address = if let Some(minter) = minter {
+                    let addr = Address::from_xdr(env, &minter)
+                        .map_err(|_| ContractError::InvalidMinter)?;
```

Comment:
Uncertain if `from_xdr` is the correct method here. I think `from_string_bytes` might be correct, because the Bytes should have already been deserialized from XDR when the message was decoded. 
https://docs.rs/soroban-sdk/22.0.0-rc.3.2/soroban_sdk/struct.Address.html#method.from_string_bytes